File: Problem Set 2/readability.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>
#include <string.h>

int main(void)
{
    float letters = 0;
    float words = 1;
    float sentences = 0;
    
    string text = get_string("Text: ");
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if ((text[i] > 64 && text[i] < 91) || (text[i] > 96 && text[i] < 123))
        {
            letters += 1;
        }
        if (text[i] == 32)
        {
            words += 1;
        }
        if (text[i] == 33 || text[i] == 46 || text[i] == 63)
        {
            sentences += 1;
        }
    }
    
    float L = (letters / words) * 100;
    float S = (sentences / words) * 100;

    int index = round((0.0588 * L) - (0.296 * S) - 15.8);
    
    if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

File: Problem Set 6/readability.py
import math

def main():
    letters = 0
    words = 1
    sentences = 0

    text = input("TEXT: ")
    n = len(text)
    for i in range(0, n):
        if 64 < ord(text[i]) < 91 or 96 < ord(text[i]) < 123:
            letters += 1
        if ord(text[i]) == 32:
            words += 1
        if ord(text[i]) == 33 or ord(text[i]) == 46 or ord(text[i]) == 63:
            sentences += 1
            
    print(f"sentences: {sentences}")
    print(f"words: {words}")
    print(f"letters: {letters}")
            
    L = (letters / words) * 100;
    S = (sentences / words) * 100;
    
    print(f"L: {L}")
    print(f"S: {S}")
    
    index = round((0.0588 * L) - (0.296 * S) - 15.8)
    
    if index > 16:
        print("Grade 16+")
    if index < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {index}")
        
main()
